File: agent/src/main/java/pt/um/lei/masb/agent/data/DataSource.java
package pt.um.lei.masb.agent.data;

import com.google.api.client.http.GenericUrl;

import java.net.URL;

public class DataSource extends GenericUrl {
    private String id;
    private URL url;

    protected DataSource() {
    }

    public DataSource(String id, URL url) {
        this.id = id;
        this.url = url;
    }

    public URL getUrl() {
        return url;
    }

    public String getID() {
        return id;
    }

}

File: agent/src/main/kotlin/pt/um/lei/masb/agent/data/DataSource.kt
package pt.um.lei.masb.agent.data

import java.net.URI


open class DataSource(val id: String, val uri: URI) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is DataSource) return false
        if (!super.equals(other)) return false

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + id.hashCode()
        return result
    }

    override fun toString(): String {
        return "DataSource(id='$id')"
    }


}
